### ISSUE TITLE
Contacts saved with rcmcarddav are no mor visible in eM Client 7.1

### ISSUE THREAD

**Issue by Offerel** [blind-coder/rcmcarddav#196]
Today eM Client 7.1 is released. But all contacts, i saved, changed or entered via ROundcube, i couldnt find in eM Client. When i re-edit such  a Contact in another App (Android), the Contact appears also in eM Client.

**Comment by Offerel** [comment]
The Problem seems to be the REV Field. rcmcardav saves the Date in this field in the Form of "REV:2017-06-15T12:42:01+02:00", which some Client, including the new Version of eM Client does not understand. However, if i change the field in the DB Backend to "REV:20170615T124201Z", the Contact was synced correctly to eM Client. I will try to change the PHP Source to use this format.

**Comment by blind-coder** [comment]
Can you change the REV: to
`REV:2017-06-15T12:42:01Z`
and see if it still works?

**Comment by Offerel** [comment]
it works to. I have entered a fresh entry via Plugin, changed the entry  in the DB Backend and synced via eM Client. The new Contact is shown in eM Client.

**Comment by blind-coder** [comment]
RFC2426 (Vcard 3.0): https://tools.ietf.org/html/rfc2426#section-3.6.4
The REV field is defined with the Z (Zulu Time) suffix and the YYYY-MM-DDTHH:mm:ssZ" format. The YYYYMMDDTHHmmssZ format is Vcard 4.0 https://tools.ietf.org/html/rfc6350#section-6.7.9
I will change the code to use the Z suffix instead of the +HH:mm suffix.

**Comment by Offerel** [comment]
meanwhile i have tryed it with:

```
	if(!$vcf) { // create fresh minimal vcard
		$vcf = new VObject\Component\VCard(
			array(
				'UID' => $save_data['cuid'],
				//'REV' => date('c'),
				'REV' => gmdate("Y-m-d\TH:i:s\Z"),
			)
		);
	} else { // update revision
		//$vcf->REV = date("c");
		$vcf->REV = gmdate("Y-m-d\TH:i:s\Z");
	}
```
in carddav_backend.php near line 1279. This seems to work for me. I haven't tested Birtdays yet.

**Comment by blind-coder** [comment]
Great! I applied that to the master branch.

**Comment by filipnavara** [comment]
We have fixed this in eM Client version 7.1.30508. The UTC dates are still preferred for compatibility reasons. Sorry for the inconvenience.

Long story: In version 7.0 we have switched to new iCalendar parser that was much more strict, but offered better foundation going forward since we added a lot of unit tests to ensure that it accepts every possible unusual iCalendar object we have encountered so far. As part of the effort, we have built vCard parser on the same foundation, starting with a vCard 4 support as a baseline. By the time of 7.1 we have added backward compatibility for vCard 3 and partially for vCard 2.1. The support for UTC offsets in vCard 3 syntax in date/timestamp values was slightly broken until version 7.1.30508. It affected Roundcube and Axigen servers as far as we know and we fixed it before pushing the version 7.1 as an update to existing users.
